apisix/admin/init.lua

```diff
@@ -237,6 +237,40 @@ local function run()
 end
 
 
+local function get_data_planes()
+    local key = "/data_plane/server_info"
+    local is_dir = true
+    local uri_segs = core.utils.split_uri(ngx.var.uri)
+    local id = uri_segs[5]
+    if id then
+        key = key .. "/" .. id
+        is_dir = false
+    end
+
+    local etcd_cli, prefix = core.etcd.get_etcd_cli()
+    key = prefix .. key
+    local res, err = etcd_cli:readdir(key)
+    if err then
+        core.log.warn("failed to do etcd readdir: ", err)
+        core.response.exit(502, {error_msg =
```

Comment:
Like https://github.com/apache/apisix/blob/09cc4946bc051452960975b4dd7436a2497df346/apisix/admin/resource.lua#L122, we usually return 503 for this case